packages/env/lib/cli.js

```diff
@@ -40,6 +40,7 @@ const withSpinner = ( command ) => ( ...args ) => {
 				spinner.fail( error.message );
 				process.exit( 1 );
 			} else if (
+				typeof error === 'object' &&
```

Comment:
In practice, does this part of the condition make any difference?

On one hand, `exitCode` likely wouldn't be assigned to anything else that we're concerned with here. And even if it were, it would probably be fine to handle it.

On the other hand, it's good to be explicit about what we're wanting to deal with, especially if we've now become aware that `error` could either be a string or an object.

But finally, on the other other hand, `typeof x === 'object'` is not the most useful since, for example `typeof null === 'object'` and `'exitCode' in null` is "Uncaught TypeError: Cannot use 'in' operator to search for 'exitCode' in null".

Whether we expect to receive a `null` here... probably not. But it's a bit odd to be concerning ourselves with the type if we're not then prepared to deal with what can possibly match that type.

I'd probably suggest one of the following:

- Use Lodash's `isPlainObject` to check the type, or at least account for null values (`error && typeof error === 'object'`
- Don't bother dealing with the type if we're assuming at this point it's going to be either a string or an object matching the form of having `exitCode` etc properties.

Reply:
I mean, the string case is handled in the `else` condition, so we need to make sure we don't run this particular case if it is a string or it will throw a TypeError. I'm not sure if it's possible to trigger this condition with null... I guess unless we `throw null`? I'm fine also adding a null check, though, that makes it more robust :)

Reply:
> make sure we don't run this particular case if it is a string or it will throw a TypeError

Oh! I guess I assumed the `in` operator would work fine when operating on a string. Apparently it does not. TIL 😄

Reply:
😁 This is why we were getting `TypeError: Cannot use 'in' operator to search for 'exitCode' in Loading composer repositories with package information`.